fix: LEAP-1957: Fix implementation of readonly behavior for controls and audio regions

Gondragos opened the pull request:
### Reason for change
This update addresses the need to enforce readonly functionality across components, ensuring that the `isReadOnly` property is respected as intended. Additionally, the features ensure that annotations in "view all" mode remain non-editable, improving the application's integrity.

### Solution
- Implemented `isReadOnly` checks across Rating controls, Taxonomy, and Audio regions, restricting user interactions appropriately.
- Updated tests to verify readonly behavior for these components.
- Introduced a helper for accessing the "view all" button in the Toolbar component.

### Testing
- End-to-end and integration tests: Covered scenarios for ensuring all annotations in "view all" mode are locked from editing and interactions.

### Risks
No significant risks have been identified, as this change enforces UI constraints without altering the core application logic.

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

Gondragos committed:
test: LEAP-1957: Add view-all mode tests for  readonly functionality

Introduced regression integration and e2e tests to ensure annotations in "view all" mode are non-editable. Covered use cases for Rating, Taxonomy, and Audio regions. Added a new helper for accessing the "view all" button in the ToolBar component.

Gondragos committed:
fix: implement proper readonly behavior for controls and audio regions

Ensure that controls respect the `isReadOnly` property across components.
- Enabled `isEditable` for Taxonomy and linked it to disabled prop.
- Updated tests to verify readonly functionality.
- Applied the `isReadOnly` check in AudioRegionModel, Rating control, and associated UI elements to restrict user interactions properly.

hlomzik commented on web/libs/editor/src/components/NewTaxonomy/NewTaxonomy.tsx:
```diff
@@ -112,7 +112,7 @@ const NewTaxonomy = ({
   // onDeleteLabel,
   options,
   // @todo implement readonly mode
```
then this `@todo` is done! 🎉

Gondragos replied:
```suggestion
```
Done

Commit:
Remove stale todo

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14637857892)

MihajloHoma committed:
Merge branch 'develop' into 'fb-LEAP-1957/view-all-tests'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14637857892

MihajloHoma committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14637857892

Gondragos committed:
fix: Adjust updateable checks in region interactions

Refined conditions to handle `updateable` state across multiple
interactions, adding ability to select region without editing it.
- Removed unnecessary `updateable` checks for mouse events where not required.
- Added conditional guardrails for edge dragging and region selection to ensure consistent behavior.
- Simplified and enhanced readability of related conditional statements.

Gondragos committed:
feat: Add view all audio selection test with per-region

Introduce a new test to validate audio region selection functionality.
- Added `audioPerRegionConfig` to support per-region choices.
- Updated the test spec to verify selection and visibility of per-region choices in the UI.

Gondragos committed:
Merge remote-tracking branch 'origin/fb-LEAP-1957/view-all-tests' into fb-LEAP-1957/view-all-tests

bmartel reviewed (approved):
Audio changes look good,  not as familiar with the other changes but based on the tests and code also look good!

Gondragos commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14748621628)
> Successfully merged:  delete mode 100644 web/libs/ui/src/tokens/layout.scss

Gondragos committed:
Merge branch 'develop' into 'fb-LEAP-1957/view-all-tests'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14748621628
